runtime/src/accounts.rs

```diff
@@ -538,36 +539,45 @@ impl Accounts {
         filter_by_address: &HashSet<Pubkey>,
         filter: AccountAddressFilter,
     ) -> Vec<(Pubkey, u64)> {
-        self.accounts_db
-            .scan_accounts(ancestors, |collector: &mut Vec<(Pubkey, u64)>, option| {
+        if num == 0 {
+            return vec![];
+        }
+        let account_balances = self.accounts_db.scan_accounts(
+            ancestors,
+            |collector: &mut BinaryHeap<Reverse<(u64, Pubkey)>>, option| {
                 if let Some(data) = option
                     .filter(|(pubkey, account, _)| {
-                        let should_include_pubkey = match filter {
-                            AccountAddressFilter::Exclude => !filter_by_address.contains(&pubkey),
-                            AccountAddressFilter::Include => filter_by_address.contains(&pubkey),
-                        };
-                        should_include_pubkey && account.lamports != 0
+                        account.lamports != 0
+                            && match filter {
+                                AccountAddressFilter::Exclude => {
+                                    !filter_by_address.contains(&pubkey)
+                                }
+                                AccountAddressFilter::Include => {
+                                    filter_by_address.contains(&pubkey)
+                                }
+                            }
                     })
-                    .map(|(pubkey, account, _slot)| (*pubkey, account.lamports))
+                    .map(|(pubkey, account, _slot)| (account.lamports, *pubkey))
```

Comment:
Nit: put `&Pubkey` in the min-heap instead of `Pubkey`. Only copy at the very end where you do that `map(|Reverse(...)|)`.

Reply:
Hmm, the borrow-checker's not too keen on that one, as the `&Pubkey` comes from `option`, which can't escape the closure:
```
`option` is a reference that is only valid in the closure body
```
Do you know any Rust tricks that might work here?

Reply:
I see.  Best we can do is only copy when adding to the collector.  I'll push a patch shortly.

Reply:
ty!